client/src/components/User/UserProfile.js

```diff
@@ -1,13 +1,143 @@
-import React, { Fragment } from "react";
+import React, { useState } from "react";
 import { useSelector, useDispatch } from "react-redux";
+import "./UserProfile.css";
+import CustomButton from "../Custom/CustomButton";
+import CustomFormInput from "../Custom/CustomFormInput";
+import { useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { auth } from "../../firebase/firebaseConfig";
+import { updateProfile } from "firebase/auth";
+import { updateUserInfo } from "../../redux/userSlice.js";
 
 function UserProfile() {
-  const user = useSelector((state) => state.user);
-  console.log(user);
+  const user = useSelector((state) => state.user.user);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const { handleSubmit, reset, register } = useForm({});
+  const changePassword = () => {
+    navigate("/forgot-password");
+  };
+  const updateUser = (event) => {
+    console.log(event);
+    const currentUser = auth.currentUser;
+    updateProfile(currentUser, {
+      displayName: event.accountholder,
+      email: event.email,
+      address: event.address,
+      phone: event.phone,
+    })
+      .then(() => {
+        dispatch(
+          updateUserInfo({
+            displayName: event.accountholder,
+            email: event.email,
+            address: event.address,
+            phone: event.phone,
+          })
+        );
+        console.log("Profile updated successfully.");
+        console.log(user);
+      })
+      .catch((error) => {
+        console.error("Error updating profile:", error);
+      });
+  };
   return (
-    <div>
-      UserProfile
-      <h1>{user.name}</h1>
+    <div className="profile_container">
+      <form>
+        <div className="profile_input">
+          <div className="image_uploader">
+            <img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQx9tjaExsY-srL4VsHNE_OKGVCJ-eIFNBktw&usqp=CAU" />
+          </div>
+          <div className="input_row">
+            <div className="profile_input_field">
+              <CustomFormInput
+                name="Account Holder"
+                id="accountholder"
+                type="text"
+                value={user?.name}
+                label=" Account Holder"
```

Comment:
Is the space before the words meant to be there? If so, you can just ignore this comment

Reply:
Yes, it is Label, I believe it looks good with space.